content/automate/ManageIQ/Transformation/TransformationHosts/Common.class/__methods__/utils.rb

```diff
@@ -128,8 +160,81 @@ def self.virtv2vwrapper_options_vmwarews2rhevm_ssh(task)
+                },
+                :osp_destination_project_id => destination_cloud_tenant.ems_ref,
+                :osp_volume_type_id         => destination_cloud_volume_type.ems_ref,
+                :osp_flavor_id              => destination_flavor.ems_ref,
+                :osp_security_groups_ids    => [destination_security_group.ems_ref],
+                :source_disks               => task[:options][:virtv2v_disks].map { |disk| disk[:path] },
+                :network_mappings           => task[:options][:virtv2v_networks],
+                :install_drivers            => true
+              }
+            end
+            private_class_method :virtv2vwrapper_options_vmwarews2openstack_vddk
+
+            def self.virtv2vwrapper_options_vmwarews2openstack_ssh(task, handle = $evm)
+              source_vm = task.source
+              source_cluster = source_vm.ems_cluster
+              source_storage = source_vm.hardware.disks.select { |d| d.device_type == 'disk' }.first.storage
+
+              destination_cloud_tenant = task.transformation_destination(source_cluster)
+              destination_ems = destination_cloud_tenant.ext_management_system
+              destination_cloud_volume_type = task.transformation_destination(source_vm.hardware.disks.select { |d| d.device_type == 'disk' }.first.storage)
+
+              destination_flavor = handle.vmdb(:flavor).find_by(:id => task.get_option(:destination_flavor_id))
+              destination_security_group = handle.vmdb(:security_group).find_by(:id => task.get_option(:destination_security_group_id))
+
+              {
+                :vm_name                    => vmware_uri_ssh(source_vm, source_storage),
+                :transport_method           => 'ssh',
+                :osp_environment            => {
+                  :os_no_cache         => true,
+                  :os_auth_url         => openstack_auth_url(destination_ems),
+                  :os_user_domain_name => destination_ems.uid_ems,
+                  :os_username         => destination_ems.authentication_userid,
+                  :os_password         => destination_ems.authentication_password,
+                  :os_project_name     => destination_cloud_tenant.name
+                },
+                :osp_destination_project_id => destination_cloud_tenant.ems_ref,
+                :osp_volume_type_id         => destination_cloud_volume_type.ems_ref,
+                :osp_flavor_id              => destination_flavor.ems_ref,
+                :osp_security_groups_ids    => [destination_security_group.ems_ref],
+                :source_disks               => task[:options][:virtv2v_disks].map { |disk| disk[:path] },
+                :network_mappings           => task[:options][:virtv2v_networks],
+                :install_drivers            => true
+              }
+            end
+            private_class_method :virtv2vwrapper_options_vmwarews2openstack_ssh
```

Comment:
Is the difference between `virtv2vwrapper_options_vmwarews2openstack_vddk` and `virtv2vwrapper_options_vmwarews2openstack_ssh` just the `:transport_method` key in the returned hash?

If so, I think it would be better to have a single method that varied on what type of `transport_method`.  I worry about someone having to go back and update this some point later and not realizing they have to change both `ssh` and `vddk` methods.

Something like:

```ruby
def virtv2vwrapper_options_vmwarews2openstack(task, transport, handle = $evm)
  ... # all the stuff that's currently in both methods
  ...
    :transport_method           => transport,
  ...
end

def virtv2vwrapper_options_vmwarews2openstack_vddk(task, handle = $evm)
  virtv2vwrapper_options_vmwarews2openstack(task, "vddk", handle)
end

def virtv2vwrapper_options_vmwarews2openstack_ssh(task, handle = $evm)
  virtv2vwrapper_options_vmwarews2openstack(task, "ssh", handle)
end
```

Reply:
There's a little bit more to it. The credentials are different. And BTW, I forgot a field for SSH transport method: `ssh_key` which contains the OpenStack provider SSH private key.

Talking about splitting code in methods, there's also work to on VMware to RHV methods. I'd rather file another PR for refactoring these. And I'm still waiting on https://github.com/ManageIQ/manageiq-content/pull/421.

Reply:
> I'd rather file another PR for refactoring these

:+1: